opentracing/README.md

```diff
@@ -0,0 +1,121 @@
+# DCP RFC
+
+For background:
+- [Main DCP doc](https://paper.dropbox.com/doc/Distributed-Context-Propagation-RGvlvD1NFKYmrJG9vGCES)
+- [DCP "Layers" doc](https://paper.dropbox.com/doc/DCP-Layers-and-Semantics-X1Tm1MSiBJECBkWYQKS2I)
+
+## API overview for those adding instrumentation
+
+Everyday consumers of this `opentracing` package really only need to worry
+about a couple of key abstractions: the `StartSpan` function, the `Span`
+interface, and golang's `context.Context` idiom. Here are code snippets
+demonstrating some important use cases.
+
+A note about `context.Context`: the opentracing API here encourages (but does
+not require) the use of golang-team's `context.Context` scheme. Whether we like
+it or not (this author is on the fence), `context.Context` is here to stay,
+goroutine-local-storage bedamned. When in Rome, do as the Romans do (etc).
+
+#### Singleton initialization
+
+The simplest starting point is `./global.go`. As early as possible, call
+
+    import ".../opentracing"
+    
+    func main() {
+        tracingRecorder := ... // tracing impl specific
+        tracingContextIDSource := ... // tracing impl specific
+        opentracing.InitGlobalTracer(tracingRecorder, tracingContextIDSource)
+        ...
+    }
+
+##### Note: the singletons are optional
+
+If global singletons make you sad, use `opentracing.NewStandardTracer(...)`
+directly and manage ownership of the `opentracing.OpenTracer` explicitly.
+
+#### Creating a Span given an existing Golang `context.Context`
+
+    func xyz(ctx context.Context, ...) {
+        ...
+        sp, ctx := opentracing.StartSpan("span_name", ctx)
```

Comment:
I was writing a long reply, and in the process convinced myself that you're right, we should not prescribe a particular way of expressing hierarchy of span _significance_. Some spans need various attributes like component, version, operation, sub-module, etc., while others may only need a name. But I would like to make one thing clear in the docs - a span with just a name is implicitly assumed to be a _local_ span. Without that I think we will create massive name clashes that the UI won't be able to resolve. For example, if both service A and service B have a local span called "orm", each with custom implementation of ORM, it doesn't make sense for the service dependency diagram to show that both A and B depend on the same "orm" component, since it's a name clash. On the other hand, if they both use Hibernate, then instrumentation can emit component=hibernate and that _would_ make sense to show in the deps diagram.

> I am 100% in favor of adding a "component" tag to spans

Do you have a suggestion for API to do that?  What I did in uber/opentracing-go was to make begin-like methods take an optional Options object containing various common concepts that we want to encode as annotations, but without exposing developers to the fact of known how to format those annotations: https://github.com/uber-common/opentracing-go/blob/master/options.go#L37 It included component name, async flag, and "peer" address to record which remote end point an RPC client span is calling (the server accordingly records address of the client who called it, both mapped to SA,CA annotations in Zipkin).